### ISSUE TITLE
Returning Oauth2 tokens

### ISSUE THREAD

**Issue by Vektrat** [Tivix/django-rest-auth#276]
Hello,

I've been struggling with this for a while, so pardon me if it is documented or I've missed it.

I have an application which uses Oauth2. However, djang-rest-auth seems prepared to return single (and most importantly, unique per user) tokens only, is that correct?

I have tried to use the _REST_AUTH_TOKEN_MODEL_ setting to set the model of the RefreshToken (currently _oauth2_provider.models.RefreshToken_, a well known Oauth2 toolkit for Django), and the _TOKEN_SERIALIZER_ setting to a serializer where I return the given RefreshToken, plus the access token assigned to it. I believe I still would need to set _REST_AUTH_TOKEN_CREATOR_ however this seems unclear to me and it appears the toolkit is managing tokens just fine - either way this setting is unneeded until I fix the issues.

**TLDR**
Even by using the settings provided to use a custom token system, they seem lacking when using non-unique-per-user tokens such as Oauth2 - the error I get is the following:

``` 
  File "/usr/local/opt/pyenv/versions/xxxxx/lib/python3.4/site-packages/rest_auth/views.py", line 83, in post
    self.login()
  File "/usr/local/opt/pyenv/versions/xxxxx/lib/python3.4/site-packages/rest_auth/views.py", line 59, in login
    self.token = create_token(self.token_model, self.user, self.serializer)
  File "/usr/local/opt/pyenv/versions/xxxxx/lib/python3.4/site-packages/rest_auth/utils.py", line 15, in default_create_token
    token, _ = token_model.objects.get_or_create(user=user)

oauth2_provider.models.MultipleObjectsReturned: get() returned more than one RefreshToken -- it returned 2!
```

Is there a workaround around this? Am I doing something wrong? Or would the project need to further evolve?
All I can think off is to create a custom proxy class that manages the token logic for me and leaves it all ready for django-rest-auth.

Thank you for your time, even with this issue, your project is saving me time :)

**Comment by Vektrat** [comment]
Regardless of this, I'll just use the Oauth2 provider API for authentication, and django-rest-auth for the rest of the calls
